### PR TITLE
(DOCSP-31727): Java SDK Migrate PBS to FS Updates

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-31727

### Staged Changes

- [Sync Data](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-31727/sdk/java/sync/): Remove PBS mentions
- [Partition-Based Sync](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-31727/sdk/java/sync/partition-based-sync/): New page with moved PBS info and PBS to FS migration info
- [Configure & Open a Synced Realm](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-31727/sdk/java/sync/configure-and-open-a-synced-realm/): Remove PBS info, update the "Synchronous Reads & Writes on the UI Thread" `Important` callout to show FS code examples

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/java/sync.txt [hunk 1]
```diff
@@ -77,20 +68,6 @@ Group Updates for Improved Performance
 
 .. include:: /includes/sync-memory-performance.rst
 
-.. _java-partition-based-sync-fundamentals:
```
Comment (cbullinger) [inline source/sdk/java/sync.txt:80]: [small nit] Can't comment on it, but does it make more sense to have the Group Updates for Improved Performance subsection under the Sync Data section?

**Review by cbullinger — APPROVED** [review]
Copy updates LGTM

File: source/sdk/java/sync.txt (hunk 1 above)
Comment (dacharyc) [reply source/sdk/java/sync.txt:80]: Makes sense - moved!

**Comment by dacharyc** [comment]
The new tests are passing locally, and the tests that are failing here in CI are passing locally. Per internal discussion, merging this PR, and will cut a separate ticket to address Java SDK test suite issues.

**Pull request merged**